### PR TITLE
sjawhar-legion-432: restore session titles in KV registry

### LINKED ISSUE #432: fix(envoy): restore session titles in KV registry — regression from PR#393
## Problem

PR #393 stripped the file registry which stored `session.title`. The KV `SessionEntry` only has port/machine/dir/updated_at — no title. Session titles disappeared from `oc ps` and `/v1/sessions`.

## Fix

1. Add `Title string json:"title"` to `SessionEntry` in `packages/envoy/internal/session/registry.go`
2. Add `title` field to the subscribe endpoint handler in `cmd/listener/main.go`
3. Plugin sends `title` in heartbeat/subscribe payload (fetch from `/session/{id}` API)
4. `/v1/sessions` endpoint includes title in response
5. `envoy` CLI script displays title column

## Context

- Regression from PR #393 (file registry removal)
- User-visible — `oc ps` no longer shows session titles
- From Envoy domain expert

## Scope

- `packages/envoy/internal/session/registry.go`
- `packages/envoy/cmd/listener/main.go`
- `packages/envoy-plugin/src/index.ts`

### PR DESCRIPTION
**Pull request by sjawhar**
Closes #432

## Summary

Restores session title display in Envoy's `/v1/sessions` API response — regression from PR #393 which removed the file registry that previously stored titles.

**Changes:**
- **Go (envoy listener):** Added `Title` field to `SessionEntry` struct, wired through the subscribe handler body and `/v1/sessions` response. New test verifies title flows through KV to the API.
- **TypeScript (envoy-plugin):** Plugin fetches session title from the serve API (`GET /session/{id}`) on session activation, caches it, and includes it in all subscribe/heartbeat payloads. Follow-up subscribe sends the title once the async fetch completes. Session-switch race guard prevents stale title writes.

### CODE REVIEW

**Comment by sjawhar** [comment]
## Behavioral Test Results

### Summary
**PASS** — 4/4 acceptance criteria verified

### Results
| Criterion | Status | Evidence |
|-----------|--------|----------|
| Add `Title` field to `SessionEntry` struct | ✅ | Verified by code inspection: `packages/envoy/internal/session/registry.go:19` — `Title string `json:"title"``. Go session registry tests pass (14/14). |
| Add `title` to subscribe endpoint handler | ✅ | Verified by code inspection: `cmd/listener/main.go:308` (body struct), `main.go:329` (passed to SessionEntry). Full listener test suite: 22/22 PASS including `TestSessionsHandler_IncludesTitle`. |
| Plugin sends `title` in heartbeat/subscribe payload | ✅ | Verified by code inspection: `index.ts:50-61` (fetchTitle from `/session/{id}`), `index.ts:88` (heartbeat), `index.ts:133` (initial subscribe), `index.ts:137-151` (follow-up subscribe with title), `index.ts:121` (race guard). Plugin tests: 22/22 PASS including title test. Plugin build: success (0.46 MB). |
| `/v1/sessions` includes title in response | ✅ | Verified by code inspection: `main.go:117` (sessionInfo struct), `main.go:147` (populated from entry). Go test `TestSessionsHandler_IncludesTitle` passes with real NATS KV — puts entry with Title="My Session Title", verifies it appears in /v1/sessions JSON response. Live Envoy confirms regression exists: `curl http://127.0.0.1:9020/v1/sessions | jq '.[0] | keys'` returns `["dir","machine_id","port","session_id","topics","updated_at"]` — no `title` field (expected: running pre-PR binary). |

### Test Evidence

**Go listener tests (22/22 PASS):**
```
cd packages/envoy && go test ./cmd/listener/ -v -count=1
PASS ok github.com/sjawhar/envoy/cmd/listener 7.718s
```

**Go session registry tests (14/14 PASS):**
```
cd packages/envoy && go test ./internal/session/... -v -count=1
PASS ok github.com/sjawhar/envoy/internal/session 11.448s
```

**Plugin build:**
```
cd packages/envoy-plugin && bun run build
Bundled 73 modules in 12ms — index.js 0.46 MB
```

**Plugin tests (22/22 PASS):**
```
cd packages/envoy-plugin && bun test
22 pass, 0 fail, 38 expect() calls [567.00ms]
```

**Live Envoy health:**
```
curl -fsS http://127.0.0.1:9020/healthz → {"status":"healthy"}
curl -fsS http://127.0.0.1:13381/session | jq length → 100
```

### Test Quality Critique
- **Go `TestSessionsHandler_IncludesTitle`**: Tests the output pipeline (KV → handler → JSON) with real NATS. Missing: no test for subscribe handler input accepting and storing `title`, no negative test for empty title. Adequate but not comprehensive.
- **TS `session title` test**: Tests async fetch → follow-up subscribe flow with mocked fetch. Missing: no race guard test (fast session switch), no heartbeat title verification, no initial-empty-then-populated assertion. Adequate but has gaps.
- **Overall**: P2 — tests verify the happy path. Safety net has gaps but core regression is covered.

### Documentation Feedback
- PR body clearly explains the regression origin (PR #393) and 2-task structure (Go plumbing + plugin).
- Issue body enumerates 5 fix points, with CLI script correctly scoped out as separate follow-up.
- No user-facing documentation changes needed — this is an internal API field addition.

### Observations
- **P3 formatting**: `gofmt` reports indentation error in `cmd/listener/main.go:308` — `Title` field in subscribe handler body struct uses single tab instead of double. Run `gofmt -w cmd/listener/main.go` to fix.
- Plugin design correctly handles the async title fetch race: initial subscribe sends empty title, follow-up subscribe sends real title after fetch completes, with session-switch guard preventing stale writes.
- The `envoy_sessions` tool description already mentions `title` in its description string — good.

### Config Compliance
- require_specific_task: false (default) — N/A
- require_taiga_evidence: false (default) — N/A

**Comment by sjawhar** [comment]
## Review Summary

**CRITICAL (P1):** 0 issues
**IMPORTANT (P2):** 1 issue
**MINOR (P3):** 1 suggestion

**Verdict: Approved** — no blocking issues. Code correctly restores session titles in the KV registry as specified in the issue.

---

### Acceptance Criteria Verification

| Requirement | Status | Evidence |
|---|---|---|
| Add `Title` to `SessionEntry` in `registry.go` | ✅ | Field added between `Dir` and `UpdatedAt`, correct JSON tag |
| Add `title` to subscribe endpoint handler in `main.go` | ✅ | Body struct updated, passed to `SessionEntry` creation |
| Plugin sends `title` in heartbeat/subscribe payloads | ✅ | `fetchTitle()` fetches from serve API, cached in `activeSessionTitle`, included in all subscribe/heartbeat calls |
| `/v1/sessions` includes title in response | ✅ | `sessionInfo` struct updated, wired through `sessionsHandler` |
| CLI script displays title column | ⏭️ | Explicitly scoped out in approved plan (separate dotfiles repo) |

### Findings

**P2 — Test coverage gaps** (not blocking for regression fix)
- No Go test for subscribe handler `Title` input (verifying Title arrives via POST body → stored in KV)
- No TS test for session-switch race guard (`activeSessionID === sessionID` check)
- No TS test for heartbeat including cached title
- Existing tests cover the happy path: Go test verifies title flows from KV → API response, TS test verifies follow-up subscribe includes title after session activation

**P3 — gofmt formatting error in subscribe handler**
- `cmd/listener/main.go:308` — `Title` field uses 2 tabs instead of 3 (confirmed by `gofmt -d`)
- Not correctness-breaking but will fail strict `gofmt` checks
- The `envoy-go` CI check was pending at review time — may catch this

### Code Quality Notes
- Session-switch race guard is correctly implemented: `if (t && activeSessionID === sessionID)` prevents stale title writes
- Best-effort `fetchTitle()` with proper timeout and error swallowing — correct for non-blocking title resolution
- Follow-up subscribe pattern (initial subscribe with empty title, then re-subscribe once title arrives) matches the plan's documented concern about async title fetch
- `envoy_sessions` tool description updated to mention title — good attention to detail

### Evidence
- CI status (at review time):
  ```
  contracts     pass    13s
  envoy-plugin  pass    12s
  lint          pass    9s
  typecheck     pass    13s
  envoy-go      pending
  test          pass    44s
  ```
- gofmt diff confirming P3:
  ```diff
  -		Title     string   \`json:"title"\`
  +			Title     string   \`json:"title"\`
  ```

File: packages/envoy/cmd/listener/main.go [hunk 1]
```diff
@@ -303,6 +305,7 @@ func main() {
 			Dir       string   `json:"dir"`
 			Topics    []string `json:"topics"`
 			Port      int      `json:"port"`
+		Title     string   `json:"title"`
```
Comment (sjawhar) [inline packages/envoy/cmd/listener/main.go:308]: **[P3/MINOR]:** Indentation error — this line uses 2 tabs but should use 3 to match the surrounding struct fields (`SessionID`, `Dir`, `Topics`, `Port`). Confirmed by `gofmt -d`. Not blocking but will fail strict `gofmt` checks.

File: packages/envoy/cmd/listener/main_test.go [hunk 2]
```diff
@@ -683,6 +683,58 @@ func TestSessionsHandler_JoinsRegistries(t *testing.T) {
 	}
 }
 
+func TestSessionsHandler_IncludesTitle(t *testing.T) {
```
Comment (sjawhar) [inline packages/envoy/cmd/listener/main_test.go:686]: **[P2/IMPORTANT]:** Test gap — this test verifies title flows from KV to API response (happy path), which is good. However, there is no corresponding test for the subscribe handler input side: verifying that when a POST to `/v1/interests/subscribe` includes a `title` field, it gets stored correctly in the session KV. The tester phase also flagged missing TS tests for the session-switch race guard and heartbeat title inclusion. Not blocking for this regression fix, but worth adding if the scope expands.
